SIPS/sip-25.md

```diff
@@ -0,0 +1,225 @@
+
+This approach allows for greater flexibility and extensibility, enabling the MetaMask ecosystem to support a wide range of blockchain networks without requiring changes to the core wallet functionality.
+
+This SIP outlines the key responsibilities and minimum interface requirements for RPC routing Snaps, enabling developers to build and deploy routing Snaps for specific blockchain networks.
+
+## Specification
+
+### Overview
+
+RPC routing Snaps will serve as intermediaries for specific [CAIP-2](https://github.com/ChainAgnostic/CAIPs/blob/main/CAIPs/caip-2.md) identifiers, registering themselves with MetaMask and managing account Snap registrations for their respective networks. They will interpret CAIP-27 requests targeting non-EVM networks made to their registered CAIP-2 identifier, route these requests, facilitate user-specified account Snap options for signing requests, and broadcast signed transactions to the network.
+
+### Key Responsibilities
+
+1. **Registration with MetaMask**:
+   - RPC routing Snaps must register with MetaMask for the CAIP-2 identifiers they can serve.
+
+2. **Account Snap Registration**:
+   - RPC routing Snaps will expose an interface for account Snaps which serve the same CAIP-2 identifier to register themselves with the routing Snap.
+   - Account Snaps must declare the signing methods they support with the routing Snap.
+
+3. **Handling CAIP-27 Requests**:
+   - RPC routing Snaps will interpret incoming CAIP-27 requests, identify the network-specific RPC request format, and route to the appropriate account Snap.
+   - If the request does not specify an account the routing Snap must provide UI for users to select from among the eligible signer Snaps for signing the request.
+   - They will broadcast the signed transactions to the respective blockchain network and return the result to MetaMask.
+   - They will also handle requests that do not require signatures by sending them directly to the network and returning the result to MetaMask.
+
+### Minimum Interface for Routing Snaps
+
+#### Registration with MetaMask
+
+- **Method**: `routingSnap_register`
+- **Description**: Registers the routing Snap with MetaMask for a specific CAIP-2 identifier(s), specifying the methods and events it can handle.
+- **Parameters**:
+  - `chains`: An array of CAIP-2 identifiers the Snap can handle.
+  - `methods`: An array JSON-RPC methods the Snap supports.
+  - `events`: An array of events the Snap can handle.
+
+#### Registering Account Snaps
```

Comment:
Seems like this should be the responsibility of the account Snap to flag as part of its permissions as well.